zetaclient/chains/base/observer.go

```diff
@@ -406,6 +406,22 @@ func (ob *Observer) PostVoteInbound(
 		logs.FieldCoinType: coinType.String(),
 	}
 
+	cctxIndex := msg.Digest()
+
+	// The cctx is created after the inbound ballot is finalized
+	// 1. if the cctx already exists, we could try voting if the ballot is present
+	// 2. if the cctx exists but the ballot does not exist, we do not need to vote
+	_, err := ob.ZetacoreClient().GetCctxByHash(ctx, cctxIndex)
+	if err == nil {
+		// The cctx exists we should still vote if the ballot is present
+		_, err = ob.ZetacoreClient().GetBallotByID(ctx, cctxIndex)
+		if err != nil {
```

Comment:
I think distinguishing `NotFound` from other errors (e.g. RPC failure) is better. Other errors don't mean the ballot deletion already happened, and the ballot may still exist and is open for vote?